### PR TITLE
[wrangler|topotools] Migrate `UpdateShardRecords`, `RefreshTabletsByShard`, and `{Get,Save}RoutingRules`

### PR DESCRIPTION
**Pull request by ajm188**
## Description

I am moving these to `package topotools` to reuse this code in both
`package grpcvtctldserver` (which wrangler imports) and `package
wrangler`.

Signed-off-by: Andrew Mason <amason@slack-corp.com>

## Related Issue(s)

- #7931 

## Checklist
- [x] Tests were added or are not required
- [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/topotools/keyspace.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+		if err != nil {
+			return err
+		}
+
+		shards[i] = updatedShard
+
+		// For 'to' shards, refresh to make them serve. The 'from' shards will
+		// be refreshed after traffic has migrated.
+		if !isFrom {
+			_ = RefreshTabletsByShard(ctx, ts, tmc, si, cells, logger)
```
Comment (rafael) [inline go/vt/topotools/keyspace.go:114]: I like this pattern of being explicit that we are ignoring the error. However, is not a pattern that I've seen in the codebase.  

I wonder if start adding here will create confusion.

**Review by rafael — COMMENTED** [review]
This looks good. Left a small comment. Let me know what you think.

File: go/vt/topotools/keyspace.go (hunk 1 above)
Comment (ajm188) [reply go/vt/topotools/keyspace.go:114]: My opinion is being that explicit about ignoring the error is a better (or at least more clear) pattern, and we've got to start somewhere! 😂  I'll do a quick temp check to see what some other folks think about this

Comment (rafael) [reply go/vt/topotools/keyspace.go]: That sounds like a plan. If we can get thumbs-up that we would like to make this a standard pattern moving forward, let's start here!

Comment (ajm188) [reply go/vt/topotools/keyspace.go]: Consensus seemed to be:
* yes, explicitly acknowledging that we're ignoring errors is better than implicitly doing so
* we've had lots of trouble debugging tests specifically due to errors being discarded
* we should log the ignored error in this specific case, and whether to log or not is probably a case-by-case basis
* i'm also going to work on [adding `errcheck` to the linter](https://github.com/vitessio/vitess/pull/7996#issuecomment-829241274) so we can start to enforce this as a pattern going forward

**Review by rafael — APPROVED** [review]
LGTM

File: go/vt/topotools/keyspace.go (hunk 1 above)
Comment (rafael) [reply go/vt/topotools/keyspace.go]: Great, thanks for following up on this. All makes sense.

**Pull request merged**